### ISSUE TITLE
Qinit unstable

### ISSUE THREAD

**Issue by snowman2** [c-h-david/rapid#4]
The results of running with Qinit are unstable. Sometimes produce reasonable results. Sometimes -inf.

**Comment by c-h-david** [comment]
Hi Alan,
Some changes were made to RAPID in January (tag 20170113) concerning the writing of a final state file (Qfinal) and the initialization using a given state file (Qinit). Are you aware of those? This perhaps caused the issue you're seeing. RAPID now uses netCDF files for both, and the sorting of river IDs is now consistent in these two files (they follow rapid_connect and m3_riv).
Let me know,
Cedric

**Comment by snowman2** [comment]
@c-h-david, I was not aware of that. This is exciting! Can you point me to some documentation/examples for this?

**Comment by snowman2** [comment]
@c-h-david, I have a quick question. Are csv files still supported?

**Comment by c-h-david** [comment]
Unfortunately, I had to remove the csv capability. That's because RAPID runs internally with double precision floats and the only way to accurately save those numbers in order to restart at the same exact model state was to use a double-precision binary file (here, netCDF).  I decided that this was important for reproducibility.  
I have yet to make documentation for the restart capability... However, you'll find an example on how to use the Qinit/Qfinal at:
[https://github.com/c-h-david/rapid/blob/master/rtk/rtk_pub_xtra_David_etal_2011_JHM.sh](url)
This is a test with a four year simulation at 15-min time step with 3-hourly outputs.  First by running all 4 years at once.  Second by running 2 years, saving the final state, and running the remaining two years while starting from that same state. The outputs should be exactly the same. 
Note that this particular test fails on Travis CI because apt-get doesn't install a recent version of NCO and concatenation of files seem to fail. But it works on my own machine.
